openlr/openlr_decoder.cpp

```diff
@@ -50,6 +69,142 @@ struct alignas(kCacheLineSize) Stats
   uint32_t m_tightOffsets = 0;
   uint32_t m_total = 0;
 };
+
+void ExpandFakes(Index const & index, Graph & g, Graph::EdgeVector & path)
+{
+  auto const isRealVertex = [&index](FeatureID const & fid, m2::PointD const & p) {
+    Index::FeaturesLoaderGuard g(index, fid.m_mwmId);
+    auto const ft = g.GetOriginalFeatureByIndex(fid.m_index);
+    bool matched = false;
+    ft->ForEachPoint(
+        [&p, &matched](m2::PointD const & fp) {
+          if (p == fp)
+            matched = true;
+        },
+        FeatureType::BEST_GEOMETRY);
+    return matched;
+  };
+
+  auto const expandFake = [&g, &path, isRealVertex](Graph::EdgeVector::iterator edgeIt) {
```

Comment:
Why not to pass ```Graph::Edge const &``` here?

Reply:
I can't pass by const ref due to this
```[c++]
if (2 * EdgeLength(*edgeIt) >= EdgeLength(*it))
  *edgeIt = *it;
else
  path.erase(edgeIt);
```
But I can pas it by non-const ref if you think it would be better.

Reply:
Using iterator is more convenient here.